### ISSUE TITLE
Breaking Change: Removal of truck speed limits and vehicle attributes from the Snap to Roads API in the Bing Maps Routes Service on 9/30/2024

### ISSUE THREAD

**Issue by ttaylor-avail** [microsoft/BingMapsRESTToolkit#81]
My apologies for posting this as a bug. I wasn't sure where to ask my question.

Our company contact received an email from Microsoft with the following title:
"Breaking Change: Removal of truck speed limits and vehicle attributes from the Snap to Roads API in the Bing Maps Routes Service on 9/30/2024"

The only link in the email just takes you to the API learning materials, which has limited info concerning the change.

I would like to know if this upcoming change in 9/30/2024 would affect this toolkit beyond the use of the parameters in the request.

**Comment by rbrundritt** [comment]
I hadn't seen that email, but by the sounds of it, this is specific to truck-based queries on snap to road API. If you are not using trucks as the travel mode with that API, this library will work fine. If you are using trucks as the travel mode, and specify vehicle attributes then it's possible the end service may throw an error (there hasn't been any disclosure on what will happen. Ideally it will simply ignore those parameters). If the API does not ignore those parameters and throws an error, then that would of course flow back to this library as the response. In regards to speed limits, I would expect that if you request that data, it simply would not be returned. Assuming that's the case, this library would continue to work fine, the only potential issue would be if your app has any code dependencies on there being speed limit data in the response.

All that said, I would fully expect that the Snap to Road API will simply ignore vehicle attributes and not return speed limits starting 9/30/2024, and this library will continue to work without issue, just without speed limit data in the response.

**Comment by ttaylor-avail** [comment]
Thank you for your response. This is basically the response I was hoping for :)

I did some due diligence, and poked around on different sites looking for more details. I also emailed the contact in the notification email we received. The response pointed me back to the email notification and the bing maps portal. Nothing specific beyond the title of the email/announcement was provided.  Well ... the link to the Microsoft learning docs for the API was provided, but it hasn't been updated in a couple years, and I didn't see any links referencing the change.

Thanks again.
